Native/NativeInputDeviceManager.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine.Experimental.Input.Utilities;
+using UnityEngine.InputNew;
+using UnityEngineInternal.Input;
+
+namespace UnityEngine.Experimental.Input
+{
+    class NativeInputDeviceManager : ScriptableSettings<NativeInputDeviceManager>
```

Comment:
Does this really need to be a scriptablesettings?  It's bringing in a bunch of non-input code and it looks like discovered devices are added dynamically to the devicerecords anyways?

Reply:
The native device manager needs to be a singleton and its records need to survive domain reload - ScriptableSettings seems like a good fit for that. Ideally we'd have some form of ScriptableSettings in Unity's API itself. But all that said, I can look into removing the dependency on non-input code.